src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -134,9 +137,28 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     accessPolicyDAO.createPolicy(AccessPolicy(resourceAndPolicyName, members, email, roles, actions))
   }
 
-  def listUserAccessPolicies(resourceType: ResourceType, userInfo: UserInfo): Future[Set[ResourceIdAndPolicyName]] = {
-    accessPolicyDAO.listAccessPolicies(resourceType.name, userInfo.userId)
-  }
+  def listUserAccessPolicies(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId): IO[Set[UserPolicyResponse]] = for{
+      rt <- IO.fromEither(resourceTypes.get(resourceTypeName).toRight(new WorkbenchException(s"missing configration for resourceType ${resourceTypeName}")))
+      isConstrained = rt.isAuthDomainConstrainable
+      ridAndPolcyName <- accessPolicyDAO.listAccessPolicies(resourceTypeName, userId) // List all policies of a given resourceType the user is a member of
+      rids = ridAndPolcyName.map(_.resourceId)
+      resources <- accessPolicyDAO.listResourceWithAuthdomains(resourceTypeName, rids)
+      authDomainMap = resources.map(x => x.resourceId -> x.authDomain).toMap
+      results = ridAndPolcyName.toList.map{
+        rnp =>
+          if(!isConstrained)
```

Comment:
I would flip the if/else: `if(isConstrained) ... else ...` It is easier to read, the `!` is too easy to miss